third_party/forked/golang/json/fields.go

```diff
@@ -28,6 +30,10 @@ func LookupPatchMetadata(t reflect.Type, jsonField string) (reflect.Type, string
 		return nil, "", "", fmt.Errorf("merging an object in json but data type is not map or struct, instead is: %s",
 			t.Kind().String())
 	}
+	if t == reflect.TypeOf(extensions.ThirdPartyResourceData{}) {
```

Comment:
This function assumes it knows how the object will be serialized based on json tags, which is not a good assumption (for types like ThirdPartyResourceData or RawExtension). That assumption should be resolved, not special-case types (this package should definitely not be importing the pkg/apis/extensions package).